src/Payments/Api.php

```diff
@@ -174,4 +175,58 @@ public function process_legacy_payment( PaymentContext $context, PaymentResult &
 		$result->set_payment_details( array_merge( $result->payment_details, $gateway_result ) );
 		$result->set_redirect_url( $gateway_result['redirect'] );
 	}
+
+	/**
+	 * Verify all dependencies of registered payment methods have been registered.
+	 * If not, remove that payment method script from the list of dependencies
+	 * of Cart and Checkout block scripts so it doesn't break the blocks and show
+	 * an error in the admin.
+	 */
+	public function verify_payment_methods_dependencies() {
+		global $wp_scripts;
+		$payment_method_scripts = $this->payment_method_registry->get_all_active_payment_method_script_dependencies();
+
+		foreach ( $payment_method_scripts as $payment_method_script ) {
+			$deps = $wp_scripts->registered[ $payment_method_script ]->deps;
+			foreach ( $deps as $dep ) {
+				if ( ! wp_script_is( $dep, 'registered' ) ) {
+					// Log and show errors.
+					error_log(  // phpcs:ignore
+						sprintf(
+							/* translators: 1: handle of the payment gateway. 2: handle of the dependency that is missing. */
+							esc_html__( 'Payment gateway with handle %1$s has been deactivated because its dependency %2$s is not registered.', 'woo-gutenberg-products-block' ),
+							esc_html( $payment_method_script ),
+							esc_html( $dep )
+						)
+					);
+					add_action(
+						'admin_notices',
```

Comment:
Is this the best place to surface errors? I am aware that notices are easy to add but I think it does add a lot of noise within admin.

I think debug log is good, and perhaps we could add an inline script to log to the console too. Those would be visible by developers and it avoids polluting admin notices.

If that works well we should do the same for the circular dependency notice too.

Reply:
Good idea! I moved this error from being a notice to a `console.error()` in a9bd620ff1a0eaa5d8af9fd3923ef0657de7f600. For now, I didn't change the circular dependency notice, we can take care of it in a follow-up and maybe even creating a util so we can easily log in the browser console from PHP code.